DNN Platform/Modules/Journal/ServicesController.cs

```diff
@@ -1,619 +1,621 @@
+        public HttpResponseMessage GetSuggestions(string keyword)
+        {
+            try
+            {
+                var findedUsers = new List<SuggestDTO>();
+                var relations = RelationshipController.Instance.GetUserRelationships(this.UserInfo);
+                foreach (var ur in relations)
+                {
+                    var targetUserId = ur.UserId == this.UserInfo.UserID ? ur.RelatedUserId : ur.UserId;
+                    var targetUser = UserController.GetUserById(this.PortalSettings.PortalId, targetUserId);
+                    var relationship = RelationshipController.Instance.GetRelationship(ur.RelationshipId);
+                    if (ur.Status == RelationshipStatus.Accepted && targetUser != null
+                        && ((relationship.RelationshipTypeId == (int)DefaultRelationshipTypes.Followers && ur.RelatedUserId == this.UserInfo.UserID)
+                                || relationship.RelationshipTypeId == (int)DefaultRelationshipTypes.Friends)
+                        && (targetUser.DisplayName.ToLowerInvariant().Contains(keyword.ToLowerInvariant())
+                                || targetUser.DisplayName.ToLowerInvariant().Contains(keyword.Replace("-", " ").ToLowerInvariant()))
+                        && findedUsers.All(s => s.userId != targetUser.UserID))
+                    {
+                        findedUsers.Add(new SuggestDTO
+                        {
+                            displayName = targetUser.DisplayName.Replace(" ", "-"),
+                            userId = targetUser.UserID,
+                            avatar = targetUser.Profile.PhotoURL,
+                            key = keyword,
+                        });
+                    }
+                }
+
+                return this.Request.CreateResponse(HttpStatusCode.OK, findedUsers.Cast<object>().Take(5));
+            }
+            catch (Exception exc)
+            {
+                Logger.Error(exc);
+                return this.Request.CreateErrorResponse(HttpStatusCode.InternalServerError, exc);
+            }
+        }
+
+        private static bool IsImageFile(string relativePath)
+        {
+            if (relativePath == null)
+            {
+                return false;
+            }
+
+            if (relativePath.Contains("?"))
+            {
+                relativePath = relativePath.Substring(
+                    0,
+                    relativePath.IndexOf("?", StringComparison.InvariantCultureIgnoreCase));
+            }
+
+            var extension = relativePath.Substring(relativePath.LastIndexOf(
+                ".",
+                StringComparison.Ordinal) + 1).ToLowerInvariant();
+            return AcceptedFileExtensions.Contains(extension);
+        }
+
+        private static bool IsAllowedLink(string url)
+        {
+            return !string.IsNullOrEmpty(url) && (url.Contains("https://") || url.Contains("http://") || url.Contains("www."));

```

Comment:
I will resolve this one in favor of recommendation by @valadas

Reply:
I _think_ I have a better solution in my suggestion, with my suggested change it would support all non-local URL schemes like:
- http
- https
- ftp
- news
- mailto

Reply:
you are leaving out the www.

Reply:
i tried this... didnt work for the www for me so I created what I made.  I will test out @valadas solution.